### PR TITLE
Flux noise/coherence analysis

### PR DESCRIPTION
**Pull request by AdriaanRol**
Please use the following template for a pull request. 

Relates to #508 .

Changes proposed in this pull request:
- A single v2 analysis that works from the point where data is presented in a table
- Clear comments on what is exactly analyzed 
- Should replace old code that is currently scattered in notebooks and spaghetti. 

Some things I wanted to do but have not yet been able to finish 
- completely kill the old analysis, can't do because I haven't yet done the data extraction automation
- automate this for pulsed experiments. I've done it using the same core functions but have not yet integrated this. 

I want to merge this so that @NielsBultink can start using the latest version.

### CODE REVIEW

File: pycqed/analysis_v2/coherence_analysis.py [hunk 1]
```diff
@@ -907,150 +1197,404 @@ def prepare_plots(self):
...
+        Qc:             coupling Q of the readout resonator
+        chi_shift       in Hz
+        path:           filepath, if provided is used for saving the plots
+
+
+
+    Input table specification:
+           Row  | Content
+        --------+--------
+            1   | dac
```
Comment (FKMalina) [inline pycqed/analysis_v2/coherence_analysis.py]: Explain what dac is. Current? Flux? Any of those?

File: pycqed/analysis_v2/coherence_analysis.py [hunk 2]
```diff
@@ -907,150 +1197,404 @@ def prepare_plots(self):
...
     """
-    val_string = par_name
-    val_string += ': {:.3g}'.format(lmfit_par.value)
-    if lmfit_par.stderr is not None:
-        val_string += r'$\pm$' + '{:.3g}'.format(lmfit_par.stderr)
+    dac, freq, T1, Tramsey, Techo, exclusion_mask = table
+    exclusion_mask = np.array(exclusion_mask, dtype=bool)
+
+    # Extract the dac arcs required for getting the sensitivities
+    fit_result_arch = fit_frequencies(dac, freq, dac0_guess=.1)
```
Comment (FKMalina) [inline pycqed/analysis_v2/coherence_analysis.py]: Fails to fit dac arc when I'm using dac that is already in flux quantum units. Needs better dac0_guess. Perhaps based on parabola fit?

File: pycqed/analysis_v2/coherence_analysis.py [hunk 3]
```diff
@@ -1,27 +1,302 @@
...
+            - fit dephasing rates
+            - determine derived quantities
+        """
+        self.fit_res = OrderedDict()
+        self.proc_data_dict = deepcopy(self.raw_data_dict)
+        pdd = self.proc_data_dict
+
+        # Extract the dac arcs required for getting the sensitivities
+        # FIXME: add a proper guess function to make the fit more robust
+        self.fit_res['dac_arc'] = fit_frequencies(pdd['dac'], pdd['freq'],
```
Comment (FKMalina) [inline pycqed/analysis_v2/coherence_analysis.py]: Fails to fit dac arc when I'm using dac that is already in flux quantum units. Needs better dac0_guess. Perhaps based on parabola fit?

File: pycqed/analysis_v2/coherence_analysis.py [hunk 4]
```diff
@@ -907,150 +1197,404 @@ def prepare_plots(self):
...
+        gamma_phi_echo      dephasing rate of echo experiment
+        verbose (int)       verbosity level
+    Returns:
+        fit_result_gammas  an lmfit fit_res object
+    """
+    # create a parameter set for the initial guess
     p = lmfit.Parameters()
     p.add('slope_ramsey', value=100.0, vary=True)
     p.add('slope_echo', value=100.0, vary=True)
     p.add('intercept', value=100.0, vary=True)
```
Comment (FKMalina) [inline pycqed/analysis_v2/coherence_analysis.py]: Should there be a lower bound of 0 on the intercept?

File: pycqed/analysis_v2/coherence_analysis.py [hunk 5]
```diff
@@ -1,27 +1,302 @@
...
+
+    Coherence_table specification:
+           Row  | Content (arrays)
+        --------+--------
+            1   | dac
+            2   | frequency (Hz)
+            3   | T1 (s)
+            4   | T2 star (s)
+            5   | T2 echo (s)
+            6   | Exclusion mask (True where data is to be excluded)
```
Comment (FKMalina) [inline pycqed/analysis_v2/coherence_analysis.py]: Figuring out to set to True in the exclusion mask is a huge pain. Additional helper plots would be great. Eg. one with that use the an index as a marker for the corresponding point. Just make sure dpi is large enough to make it readable. Or use vector format.

File: pycqed/analysis_v2/coherence_analysis.py [hunk 6]
```diff
@@ -1,27 +1,302 @@
...
+
+    Generates 8 plots:
+        > T1, T2, Echo vs flux
+        > T1, T2, Echo vs frequency
+        > T1, T2, Echo vs flux sensitivity
+        > ratio Ramsey/Echo vs flux
+        > ratio Ramsey/Echo vs frequency
+        > ratio Ramsey/Echo vs flux sensitivity
+        > Dephasing rates Ramsey and Echo vs flux sensitivity
+        > Dac arc fit, use to assess if sensitivity calculation is correct.
```
Comment (FKMalina) [inline pycqed/analysis_v2/coherence_analysis.py]: Add description of where in the analysis object the relevant parameters are stored. Half of the time when I use analysis object I don't care about the plots, only about fitted parameters. Say I want to use \sqrt(A) for ramsey. I'd rather not dig through the code to figure out how to get access to that in my analysis notebook.

File: pycqed/analysis_v2/coherence_analysis.py (hunk 4 above)
Comment (AdriaanRol) [reply pycqed/analysis_v2/coherence_analysis.py]: yes.

File: pycqed/analysis_v2/coherence_analysis.py (hunk 6 above)
Comment (FKMalina) [reply pycqed/analysis_v2/coherence_analysis.py]: (Philosophical comment) This is one of the reasons why everyone writes their own scripts. It is less frustrating and easier to write your own script then to figure out where in the analysis object the results are stored.

File: pycqed/analysis_v2/coherence_analysis.py (hunk 5 above)
Comment (AdriaanRol) [reply pycqed/analysis_v2/coherence_analysis.py]: I think that is a good addition.

File: pycqed/analysis_v2/coherence_analysis.py (hunk 6 above)
Comment (AdriaanRol) [reply pycqed/analysis_v2/coherence_analysis.py]: Excellent point.

**Pull request merged**